mlflow/langchain/__init__.py

```diff
@@ -682,6 +684,67 @@ def _convert_ndarray_to_list(data):
             f"for model {self.lc_model.__class__.__name__}"
         )
 
+    def predict_stream(
+        self,
+        data: Any,
+        params: Optional[Dict[str, Any]] = None,
+    ) -> Iterator[str]:
+        """
+        Args:
+            data: Model input data, only single input is allowed.
+            params: Additional parameters to pass to the model for inference.
+
+                .. Note:: Experimental: This parameter may change or be removed in a future
+                    release without warning.
+        Returns:
+            An iterator of model prediction chunks.
+        """
+        from mlflow.langchain.api_request_parallel_processor import process_stream_request
+
+        # TODO: Q: shall we put `callback_handlers` in params ?
+        if isinstance(data, list):
+            raise MlflowException("Langchain model predict_stream only supports single input.")
+
+        data = _convert_ndarray_to_list(data)
+        return process_stream_request(
+            lc_model=self.lc_model,
+            request_json=data,
+        )
+
+    def _predict_stream_with_callbacks(
+        self,
+        data: Any,
+        params: Optional[Dict[str, Any]] = None,
+        callback_handlers=None,
+        convert_chat_responses=False,
+    ) -> Iterator[str]:
```

Comment:
Is `Iterator[str]` always correct here? My understanding is this is the default value, but when `convert_chat_responses=True`, a Dict should be returned. Separately, do we plan to eventually merge `convert_chat_responses` back into `predict / predict_stream` methods? (for tech debt consolidation)

Also, from the DD my understanding was we would return a ChatChunk object when `convert_chat_responses=True` - will it be a Dict now?
cc: @BenWilson2

Reply:
no. it might be a dict too. I will update it.